File: SchoolWebApp-Client/src/app/cbe/assessments/models/theme.ts
import { Curriculum } from '@/academics/models/curriculum';
import { Subject } from '@/academics/models/subject';
import { LearningLevel } from '@/class/models/learning-level';
import {ResourceModel} from '@/core/models/ResourceModel';

export class Theme extends ResourceModel<Theme> {
    public name?: string;
    public code?: string;
    public description?: string;
    public rank?: number;

    public curriculumId?: number;
    public curriculum?: Curriculum;

    public learningLevelId?: number;
    public learningLevel?: LearningLevel;

    public subjectId?: number;
    public subject?: Subject;

    constructor(model?: Partial<Theme>) {
        super(model);
    }
}

File: SchoolWebApp-API/Project.Core/Entities/CBE/Assessments/Theme.cs
using SchoolWebApp.Core.Entities.Academics;
using SchoolWebApp.Core.Entities.Class;
using SchoolWebApp.Core.Entities.Shared;

namespace SchoolWebApp.Core.Entities.CBE.Assessments
{
    public class Theme : SettingsBase
    {
        public string? Code { get; set; }

        public int? CurriculumId { get; set; }
        public Curriculum? Curriculum { get; set; } = null;

        public int SubjectId { get; set; }
        public Subject? Subject { get; set; } = null;

        public int LearningLevelId { get; set; }
        public LearningLevel? LearningLevel { get; set; } = null;

        public List<Strand> Strands { get; set; } = new();
    }
}
